src/nix/export.cc

```diff
@@ -0,0 +1,126 @@
+#include "command.hh"
+#include "common-args.hh"
+#include "shared.hh"
+#include "store-api.hh"
+#include "fs-accessor.hh"
+
+using namespace nix;
+
+struct CmdExport : InstallableCommand
```

Comment:
The name `export` is pretty vague and overloaded (e.g. we already have a `nix-store --export`). It's worth noting that this command does not actually export anything by itself since the build result is in the Nix store.

Maybe `bundle` (though that's not really correct either)? 

Or an "apply" flag to `nix build` analogous to `nix eval --apply`, e.g. `nix build --apply-generator github:matthewbauer/nix-bundle hello`.

Reply:
Yeah, I thought export would imply it doesn't just have to be nix-bundle proper, but it is a little confusing. I suppose you could make an exporter / bundler that just produces a .nar though.